core/src/main/java/hudson/ClassicPluginStrategy.java

```diff
@@ -88,6 +88,12 @@ public boolean accept(File dir,String name) {
         }
     };
 
+    /**
+     * If set this should be the absolute file path to the the base directory for all exploded .hpi/.jpi plugins.
+     * If {@literal null} then plugins will be exploded into {@literal JENKINS_HOME/plugins}.
+     */
+    private static final String WORK_DIR = System.getProperty(ClassicPluginStrategy.class.getName() + ".WORK_DIR");
```

Comment:
@oleg-nenashev I am not saying you are not right :laughing: but I have seen this construction somewhere else